components/Chat/UserMessage.tsx

```diff
@@ -1,22 +1,29 @@
 import type { Message } from "@ai-sdk/react";
+import Markdown from "react-markdown";
+import remarkGfm from 'remark-gfm';
+import styles from "./markdown.module.css";
 
 interface UserMessageProps {
   message: Message;
 }
 
 const UserMessage = ({ message }: UserMessageProps) => {
   return (
-    <div className="grow flex justify-end max-w-[90%]">
-      <section>
-        <div
-          className="text-sm font-sans max-w-[500px] text-pretty break-words bg-grey px-4 p-2 rounded-full"
-          dangerouslySetInnerHTML={{
-            __html: decodeURIComponent(
-              message.content.replaceAll("%", "&#37;") || ""
-            ),
-          }}
-        />
-      </section>
+    <div className={styles.markdown}>
+      <Markdown
+        remarkPlugins={[remarkGfm]}
+        components={{
+          a: ({ ...props }) => (
+            <a 
+              {...props} 
+              target="_blank" 
+              rel="noopener noreferrer"
+            />
+          )
+        }}
+      >
+        {decodeURIComponent(message.content.replaceAll("%", "&#37;") || "")}
+      </Markdown>
```

Comment:
This same block of code is repeated in BOTH
- components/Chat/UserMessage.tsx and
- components/Chat/TextMessagePart.tsx 
Could you please DRY your code by creating a new component?
- name: `<ChatMarkdown />` 
- props: `children`